src/ts/services/adapters/test/TyperighterChunkedAdapter.spec.ts

```diff
@@ -0,0 +1,170 @@
+import { noop } from "lodash";
+import { ReadableStream } from "stream/web";
+import { TextEncoder } from "util";
+import { IMatcherResponse } from "../../../interfaces/IMatch";
+import TyperighterChunkedAdapter, {
+  readJsonSeqStream,
+  RECORD_SEPARATOR
+} from "../TyperighterChunkedAdapter";
+
+const textEncoder = new TextEncoder();
+
+const createJsonSeqArrFromRecords = (records: any[]) =>
+  records.map(record => JSON.stringify(record) + RECORD_SEPARATOR);
+
+const createStream = (records: string[]) =>
+  new ReadableStream({
+    start: controller => {
+      records.forEach(record => controller.enqueue(textEncoder.encode(record)));
+      controller.close();
+    }
+  });
+
+describe("TyperighterChunkedAdapter", () => {
+  const localFetch = global.fetch;
+  const onReceived = jest.fn();
+  const onError = jest.fn();
+  const onComplete = jest.fn();
+
+  const mockFetchBody = (partialRequest: Partial<Response>) =>
+    (global.fetch = jest.fn(() => Promise.resolve(partialRequest) as any));
+
+  afterAll(() => {
+    global.fetch = localFetch;
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it("on a successful response, it should call `onMatchesReceived` for each record, and finally call `onRequestComplete`", async () => {
+    const adapter = new TyperighterChunkedAdapter("https://example.com");
+    const input: IMatcherResponse[] = [
+      {
+        blocks: [],
+        categoryIds: ["1"],
+        matches: [],
+        requestId: "request-id"
+      },
+      {
+        blocks: [],
+        categoryIds: ["2"],
+        matches: [],
+        requestId: "request-id"
+      }
+    ];
+    const jsonSeq = createJsonSeqArrFromRecords(input);
+    const stream = createStream(jsonSeq);
+    mockFetchBody({ body: stream });
+
+    await adapter.fetchMatches(
+      "request-id",
+      [],
+      [],
+      onReceived,
+      onError,
+      onComplete
+    );
+
+    expect(onReceived).toHaveBeenCalledWith(input[0]);
+    expect(onReceived).toHaveBeenCalledWith(input[1]);
+    expect(onComplete).toHaveBeenCalled();
+  });
+
+  it("on an empty response, it should call `onRequestError`, and finally call `onRequestComplete`", async () => {
```

Comment:
As above:
```
As above:
```suggestion
  it("should call `onRequestError` for an empty response, and finally call `onRequestComplete`", async () => {
```